### ISSUE TITLE
Style modulation layers, style parameters and controlling low-level features.

### ISSUE THREAD

**Issue by 007prateekd** [mchong6/JoJoGAN#34]
Hey, great work! 
I had a couple of queries:
1.  In the paper it is mentioned that there are 26 style modulation layers, but in the code it seems to be 18 as `n_latent` = 18.
2. What exactly do the style parameters _s(w)_ correspond to in the code?
3. For the pertained styles, is there any way to control low-level features like eyes, nose, etc. without training again? I know that while fine-tuning we can use blending (using RIS) and different masks for controlling them but is there any way for a model which is already fine-tuned?
4. I see a change in results when fine-tuning the model using <a href="https://github.com/mchong6/JoJoGAN/blob/main/style_images_aligned/jojo.png">JoJo's photo</a>. Is it because of _e4e_ being used instead of _ReStyle_ in the code?

**Comment by mchong6** [comment]
1) Some layers share the same latent code (like torgb layers) thus even though there are 18 latent codes, there are 26 style modulation layers.
2) It corresponds to the output of this https://github.com/mchong6/JoJoGAN/blob/ff33d3f541a8deccc6df85f053286ac8c88f7526/model.py#L237
3) A way to do that is to load the original model and fine interpolate between the finetuned/original features as described in the feature interpolation section of the paper. To specifically control eyes/nose/etc you can specifically interpolate those section by cropping those spatial regions within the features. Look at https://arxiv.org/abs/2111.01619
4) There have been several versions with several different hyperparameters. Thus you might not get the exact same result.

**Comment by 007prateekd** [comment]
1. So if I understood it correctly, then there are 17 `StyledConv` layers with independent codes and 9 `ToRGB` layers with shared codes. Hence, 26 layers and 18 codes in total.
2. Quoting your paper,

    > We GAN invert the reference style image y to obtain a style code w = T(y) and from that a set of s parameters s(w).

   So, by obtaining _s(w)_, do you mean implicitly obtaining them through that line of code you mentioned? Because in the fine-tuning code there doesn't seem to be any explicit reference to obtain these parameters.
4. Great. Will definitely give it a shot.
5. That makes sense.
